### ISSUE TITLE
Prokaryote issues: uniprot_strata(from=1) error and use_recommended_prokaryotes warnings

### ISSUE THREAD

**Issue by LTibbs** [arendsee/phylostratr#34]
I am having some trouble selecting prokaryotes to use in phylostratr.

Specifically, while following the _Arabidopsis_ vignette, there were warnings that 31 out of the ~85 prokaryotes in `use_recommended_prokaryotes` have issues ("Could not find representative proteome...", see bottom for full warnings). When I checked these problem IDs in UniProt, they had issues ranging from no results (e.g., ID 234267) to proteome status in UniProt of redundant (e.g., 617121) or excluded (e.g., ID 469616) to simply not having a reference proteome designated (e.g., ID 797303).
```r
strata <-
  # Get stratified relatives represented in UniProt
  uniprot_strata(focal_taxid, from=2) %>%
  # Select a diverse subset of 5 or fewer representatives from each stratum.
  strata_apply(f=diverse_subtree, n=5, weights=weights) %>%
  # Use a prebuilt set of prokaryotic species
  use_recommended_prokaryotes %>%
  # Add yeast and humans
  add_taxa(c('4932', '9606')) %>%
  # Download genomes, storing the filenames
  uniprot_fill_strata

There were 32 warnings (use warnings() to see them)
```

Because so many of the suggested prokaryotes had warnings, I wanted to select alternative prokaryotes. So, I tried to use `uniprot_strata` with `from=1`, which according to the documentation I thought should build a strata object from cellular organisms onward, so that then I could pipe that object to `strata_apply`. However, I get the following error from `uniprot_strata`:

```r
> strata <- uniprot_strata(focal_taxid,
+                          from=1) 
The focal species is not present in UniProt. You may add it after retrieving uniprot sequences (i.e. with 'uniprot_fill_strata') with a command such as: strata_obj@data$faa[[focal_taxid]] <- '/path/to/your/focal-species.faa'
Error in integer(max(oldnodes)) : vector size cannot be infinite
In addition: Warning message:
In max(oldnodes) : no non-missing arguments to max; returning -Inf
```
Although the error seems to say that the issue is with the focal species, I am able to run the identical command (same focal_taxid) with `from=2` with no issues.

I would appreciate any insight on how to build an updated prokaryote tree for phylostratr. Thank you!

```r
# Full warnings from use_recommended_prokaryotes:
strata <-
  # Get stratified relatives represented in UniProt
  uniprot_strata(focal_taxid, from=2) %>%
  # Select a diverse subset of 5 or fewer representatives from each stratum.
  strata_apply(f=diverse_subtree, n=5, weights=weights) %>%
  # Use a prebuilt set of prokaryotic species
  use_recommended_prokaryotes %>%
  # Add yeast and humans
  add_taxa(c('4932', '9606')) %>%
  # Download genomes, storing the filenames
  uniprot_fill_strata

There were 32 warnings (use warnings() to see them)
warnings()
> warnings()
Warning messages:
  1: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 4932, retrieving all records, may be incomplete or contain redundant entries
2: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1797576, retrieving all records, may be incomplete or contain redundant entries
3: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1184387, retrieving all records, may be incomplete or contain redundant entries
4: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 234267, retrieving all records, may be incomplete or contain redundant entries
5: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 469616, retrieving all records, may be incomplete or contain redundant entries
6: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1805218, retrieving all records, may be incomplete or contain redundant entries
7: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1703384, retrieving all records, may be incomplete or contain redundant entries
8: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1797384, retrieving all records, may be incomplete or contain redundant entries
9: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1797220, retrieving all records, may be incomplete or contain redundant entries
10: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1628278, retrieving all records, may be incomplete or contain redundant entries
11: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1262851, retrieving all records, may be incomplete or contain redundant entries
12: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1805232, retrieving all records, may be incomplete or contain redundant entries
13: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1703396, retrieving all records, may be incomplete or contain redundant entries
14: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 357808, retrieving all records, may be incomplete or contain redundant entries
15: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 573058, retrieving all records, may be incomplete or contain redundant entries
16: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1917462, retrieving all records, may be incomplete or contain redundant entries
17: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1262887, retrieving all records, may be incomplete or contain redundant entries
18: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 617121, retrieving all records, may be incomplete or contain redundant entries
19: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1956166, retrieving all records, may be incomplete or contain redundant entries
20: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1896212, retrieving all records, may be incomplete or contain redundant entries
21: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 481448, retrieving all records, may be incomplete or contain redundant entries
22: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1262865, retrieving all records, may be incomplete or contain redundant entries
23: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 199, retrieving all records, may be incomplete or contain redundant entries
24: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 48, retrieving all records, may be incomplete or contain redundant entries
25: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1798435, retrieving all records, may be incomplete or contain redundant entries
26: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 634771, retrieving all records, may be incomplete or contain redundant entries
27: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1660162, retrieving all records, may be incomplete or contain redundant entries
28: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1803813, retrieving all records, may be incomplete or contain redundant entries
29: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 1775755, retrieving all records, may be incomplete or contain redundant entries
30: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 797303, retrieving all records, may be incomplete or contain redundant entries
31: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 419665, retrieving all records, may be incomplete or contain redundant entries
32: In uniprot_retrieve_proteome_table(taxid) :
  Could not find representative proteome for 68525, retrieving all records, may be incomplete or contain redundant entries

```

**Comment by LTibbs** [comment]
Update: I forked the repository and updated `uniprot_strata()` to run successfully with `from=1`. 

The issue looks like it occurred in the `query_sparql` function, where taxon 131567 (cellular organisms) does not find any results, I think because this taxon is categorized as "no rank" in UniProt. So instead, I edited the `uniprot_strata()` function so that it will search the three sub-classes of cellular organisms (bactera, archaea, and eukarya), rather than trying to search cellular organisms as a whole, when from=1.

However, I now realize the fix of this function does not address my original motivation of creating an updated prokaryote tree, as the tree resulting from `uniprot_strata()` only contains `n` (e.g., `n=5`) of each of bacteria and archaea, which of course makes sense as it's a focused tree. So now I will work on an updated un-focused tree for prokaryotes.

**Comment by LTibbs** [comment]
Update: I created functions to build trees based on current UniProt taxonomy using the `diverse_subtree` function of phylostratr. The trees can be customized to use the desired number of species and also accept custom weights. 

I hope my solution to my issue can help others! I'll submit a pull request to hopefully make these updates available to others.

Edit: Link here: https://github.com/ltibbs/phylostratr

**Comment by arendsee** [comment]
@LTibbs Thanks for the issue report!

The IDs, such as 234267, that are raising warnings are NCBI taxonomy ids for the various bacterial taxa. At the time `phylostratr` was published, UniProt had reference proteomes for all of them. The old list of 85 strains was made by the function `uniprot_sample_prokaryotes`, which retrieved all NCBI taxonomy ids for which a UniProt representative proteome was defined and then from the ids built a tree following the NCBI taxonomy.

The warnings are mostly harmless, actually. If the "representative proteome" annotation was removed by the UniProt curators, then `phylostratr` will pull all available sequence for the given NCBI taxonomy id rather than limiting data to the defined proteome set. This can lead to duplicate data (if there are many proteomes for the taxon) or it can lead to incomplete data (if the proteome is partial). When I run this code, `phylostratr` retrieves protein data of roughly the expected size (a few thousand proteins) for almost all the bacteria. Almost. There are a few bacteria for which retrieval failed ... I haven't dug into the reason.

When I rerun `uniprot_sample_prokaryotes`, I now get 136 prokaryotic taxa, rather than 85 I retrieved back 2018. I may add this as a new prokaryotic recommended list.

Anyway, that is just a little context, I'll move further discussion to the PR request.
